BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -640,19 +683,32 @@ - (NSURLSessionTask *)backgroundTaskWithRequest:(NSURLRequest *)request orUpload
     return sessionTask;
 }
 
-- (BOOL)persistBackgroundSessionTaskWithSessionId:(NSString *)backgroundSessionId sessionTask:(NSURLSessionTask *)task taskDelegate:(id <BOXURLSessionDownloadTaskDelegate>)taskDelegate userId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error
+- (BOOL)persistBackgroundSessionTaskWithSessionId:(NSString *)backgroundSessionId
+                                      sessionTask:(NSURLSessionTask *)task
+                                     taskDelegate:(id <BOXURLSessionTaskDelegate>)taskDelegate
```

Comment:
I changed the `taskDelegate` parameter type from `id <BOXURLSessionDownloadTaskDelegate>` to `id <BOXURLSessionTaskDelegate>`. It seems this was intended to use the generic protocol, I don't see any logic specific to downloads here. Let me know if I'm mistaken.